Homework2/src/problem2/Basket.java

```diff
@@ -0,0 +1,37 @@
+package problem2;
+
+
+import java.util.ArrayList;
+
+public class Basket {
+    private double totalPrice = 0;
+    private double price = 0;
+    private Customer customer;
+    private ArrayList<Product> products = new ArrayList<>();
+
+    public Basket(Customer customer){
+        this.customer = customer;
+        totalPrice = 0;
+    }
+
+    public double getTotalPrice() {
+        totalPrice = price;
+        if(customer.getMembership() == Customer.Membership.GOLD){
```

Comment:
e ok, dar ar fi si mai bie ca in enum sa tii si discontul asociat, prin existenta acolo a unui field discount; astfel ai avea un loc centralizat unde ai pune aceasta relatie intre membership si discount si ai putea sa te folosesti de ea in mai multe locuri din codul tau cand va fi nevoie si asa va fi si usor de modificat pe viitor, vei schimba discountul doar in enum daca va fi nevoie si nu te vei atinge de alte bucati de cod